src/otx/core/data/entity/action_classification.py

```diff
@@ -72,6 +72,12 @@ def collate_fn(cls, entities: list[ActionClsDataEntity]) -> ActionClsBatchDataEn
             labels=[entity.labels for entity in entities],
         )
 
+    def pin_memory(self) -> ActionClsBatchDataEntity:
+        """Pin memory for member tensor variables."""
+        super().pin_memory()
+        self.labels = [label.pin_memory() for label in self.labels]
+        return self
+
```

Comment:
It seems ActionDetBatchDataEntity doesn't have this part

Reply:
Fixed in https://github.com/openvinotoolkit/training_extensions/pull/2749/commits/072f92dc6c8f36d67828a7cc5efe1ddbb74eacfe